File: java/Facility.java
class Facility extends Cell{
    int wellx = 14;
    int welly = 11;
    int MixerX = 14;
    int MixerY = 13;
    int TruckX = 14;
    int TruckY = 14;

    /*Constructor untuk facility*/
    public Facility() {
        super.setElement(getWellX(), getWellY(), 'W');
        super.setElement(getMixerX(), getMixerY(), 'M');
        super.setElement(getTruckX(), getTruckY(), 'T');
    }

    public Facility(Cell c) {
        c.setElement(getWellX(), getWellY(), 'W');
        c.setElement(getMixerX(), getMixerY(), 'M');
        c.setElement(getTruckX(), getTruckY(), 'T');
    }
    /*Mengembalikan letak x dari Well*/
    public int getWellX() {
        return wellx;
    }

    /*Mengembalikan letak y dari Well*/
    public int getWellY() {
        return welly;
    }

    /*Mengembalikan letak x dari Well*/
    public int getMixerX() {
        return MixerX;
    }

    /*Mengembalikan letak y dari Mixer*/
    public int getMixerY() {
        return MixerY;
    }

    /*Mengembalikan letak x dari Mixer*/
    public int getTruckX() {
        return TruckX;
    }

    /*Mengembalikan letak y dari Well*/
    public int getTruckY() {
        return TruckY;
    }

}

File: cpp/src (copy)/Facility.cpp
#include <iostream>
#include "../include/Facility.h"
using namespace std;


int Facility::wellx = 14;
int Facility::welly = 11;
int Facility::MixerX = 14;
int Facility::MixerY = 13;
int Facility::TruckX = 14;
int Facility::TruckY = 14;

/*Constructor untuk facility*/
Facility::Facility() : Cell() {
    Cell::setElement(getWellX(), getWellY(), 'W');
    Cell::setElement(getMixerX(), getMixerY(), 'M');
    Cell::setElement(getTruckX(), getTruckY(), 'T');
}

/*Mengembalikan letak x dari Well*/
int Facility::getWellX() {
    return wellx;
}

/*Mengembalikan letak y dari Well*/
int Facility::getWellY() {
    return welly;
}

/*Mengembalikan letak x dari Well*/
int Facility::getMixerX() {
    return MixerX;
}

/*Mengembalikan letak y dari Mixer*/
int Facility::getMixerY() {
    return MixerY;
}

/*Mengembalikan letak x dari Mixer*/
int Facility::getTruckX() {
    return TruckX;
}

/*Mengembalikan letak y dari Well*/
int Facility::getTruckY() {
    return TruckY;
}


